dbt/adapters/bigquery/python_submissions.py

```diff
@@ -34,7 +34,7 @@ def __init__(self, parsed_model: Dict, credential: BigQueryCredentials) -> None:
         self.credential = credential
         self.GoogleCredentials = BigQueryConnectionManager.get_credentials(credential)
         self.storage_client = storage.Client(
-            project=self.credential.database, credentials=self.GoogleCredentials
+            project=self.credential.execution_project, credentials=self.GoogleCredentials
```

Comment:
Looks like this will be default to [database](https://github.com/dbt-labs/dbt-bigquery/blob/61ed3b02a999995fe547dcc263ad1e0fd7e575f8/dbt/adapters/bigquery/connections.py#L180).